src/java.base/share/classes/javax/crypto/JceSecurityManager.java

```diff
@@ -100,11 +102,15 @@ CryptoPermission getCryptoPermission(String alg) {
         // javax.crypto.* packages.
         // NOTE: javax.crypto.* package maybe subject to package
         // insertion, so need to check its classloader as well.
-        Class<?>[] context = getClassContext();
+        PrivilegedAction<StackWalker> pa =
+                () -> StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE);
+        @SuppressWarnings("removal")
+        List<StackFrame> stack =
+                AccessController.doPrivileged(pa).walk(Stream::toList);
```

Comment:
You can replace line 108-125 with something like this: 
```
StackWalker walker = AccessController.doPrivileged(pa);
Optional<URL> callerCodeBase = walker.walk(s -> {
    s.map(f -> JceSecurity.getCodeBase(f.getDeclaringClass()))
      .findFirst();
});
```

Reply:
@mlchung Maybe there should be a `.filter(cb -> cb != null)` inserted before `.findFirst()`?

Reply:
Daniel is right and it needs the filter to find the first non-null code base.